### PR TITLE
2146 - refactoring to a generic endpoiont

### PR DESCRIPTION
**Pull request by codyseibert**

### CODE REVIEW

File: shared/src/business/useCases/generatePdfFromHtmlInteractor.js [hunk 1]
```diff
@@ -56,9 +60,9 @@ exports.generateDocketRecordPdfInteractor = async ({
 
     result = await page.pdf({
       displayHeaderFooter: true,
-      footerTemplate,
+      footerTemplate: showFooter ? footerTemplate : '',
```
Comment (matthopson) [inline shared/src/business/useCases/generatePdfFromHtmlInteractor.js]: Why not use the `displayHeaderFooter` bool above?

Comment (matthopson) [reply shared/src/business/useCases/generatePdfFromHtmlInteractor.js]: Maybe this gives us better control over which to display?

File: web-api/swagger.json [hunk 2]
```diff
@@ -2626,7 +2626,7 @@
         }
       }
     },
-    "/docket-record-pdf": {
+    "/generate-pdf-from-html": {
       "post": {
         "summary": "generate a printable pdf of the docket record",
```
Comment (matthopson) [inline web-api/swagger.json]: Update summary?

Comment (matthopson) [reply web-api/swagger.json]: And desc.

**Review by matthopson — APPROVED** [review]

File: shared/src/business/useCases/generatePdfFromHtmlInteractor.js [hunk 3]
```diff
@@ -55,10 +57,10 @@ exports.generateDocketRecordPdfInteractor = async ({
     `;
 
     result = await page.pdf({
-      displayHeaderFooter: true,
-      footerTemplate,
+      displayHeaderFooter,
+      footerTemplate: footerTemplate,
```
Comment (kkoskelin) [inline shared/src/business/useCases/generatePdfFromHtmlInteractor.js:61]: why change this (and headerTemplate) from the original (and simpler) `footerTemplate,` ?

File: shared/src/business/useCases/generatePdfFromHtmlInteractor.js [hunk 4]
```diff
@@ -1,14 +1,16 @@
 /**
- * generateDocketRecordPdfInteractor
+ * generatePdfFromHtmlInteractor
```
Comment (kkoskelin) [inline shared/src/business/useCases/generatePdfFromHtmlInteractor.js:2]: Perhaps you will also want to alter the string sent to `applicationContext.logger.time` on line 22?

File: web-api/src/cases/generatePdfFromHtmlLambda.js [hunk 5]
```diff
@@ -51,15 +51,19 @@ exports.handler = event =>
   customHandle(event, async () => {
     const user = getUserFromAuthHeader(event);
     const applicationContext = createApplicationContext(user);
-    const { docketNumber, docketRecordHtml, headerHtml } = JSON.parse(
-      event.body,
-    );
+    const {
+      displayHeaderFooter,
+      docketNumber,
+      docketRecordHtml,
```
Comment (kkoskelin) [inline web-api/src/cases/generatePdfFromHtmlLambda.js]: Perhaps this variable should also be renamed to be more generic?

**Review by kkoskelin — CHANGES_REQUESTED** [review]
Perhaps a few things that could be adjusted to clarify intent.

File: web-api/src/cases/generatePdfFromHtmlLambda.js (hunk 5 above)
Comment (matthopson) [reply web-api/src/cases/generatePdfFromHtmlLambda.js]: We were just discussing some of this in Slack.

**Review by kkoskelin — APPROVED** [review]

**Review by sutt0n — APPROVED** [review]

**Pull request merged**